pages/api/data-sources/[dataSourceId]/tables/[tableName]/records/[recordId].ts

```diff
@@ -104,19 +104,30 @@ async function handlePUT(req: NextApiRequest, res: NextApiResponse) {
 
   const service = await getQueryService({ dataSource });
 
-  const data = await service.runQuery("updateRecord", {
-    tableName: req.query.tableName as string,
-    recordId: req.query.recordId as string,
-    data: req.body.changes,
-  });
+  const [record, data]: [Record<string, any>, Promise<unknown>] =
+    await service.runQueries([
+      {
+        name: "getRecord",
+        payload: {
+          tableName: req.query.tableName as string,
+          recordId: req.query.recordId as string,
+        },
+      },
+      {
+        name: "updateRecord",
+        payload: {
+          tableName: req.query.tableName as string,
+          recordId: req.query.recordId as string,
+          data: req.body.changes,
+        },
+      },
+    ]);
```

Comment:
This block is a bit different than `[main](https://github.com/basetool-io/basetool/blob/main/pages/api/data-sources/[dataSourceId]/tables/[tableName]/records/[recordId].ts)`.
Can you make sure we don't roll back any changes?